### PR TITLE
Use custom LLVM

### PR DESCRIPTION
**Pull request by jackcmay**
#### Problem

Solana requires modifications to LLVM that have not been upstreamed. 

#### Summary of Changes

Customized LLVM repos are being maintained by the Solana Github organization and built/released via the repo: https://github.com/solana-labs/llvm-builder

The changes in this PR switches Solana over to using the custom LLVM which includes downloading and installing the release from llvm-builder during sdk install.

docker-llvm was maintained in the Solana repo, it has been moved to llvm-builder repo.

Fixes #

### CODE REVIEW

File: programs/bpf/c/sdk/scripts/install.sh [hunk 1]
```diff
@@ -4,20 +4,51 @@ cd "$(dirname "$0")"/..
...
 version=v2.3.2
-(
-  [[ ! -d criterion-$version ]] || exit 0
-  set -ex
-  wget https://github.com/Snaipe/Criterion/releases/download/$version/criterion-$version-osx-x86_64.tar.bz2
-  wget https://github.com/Snaipe/Criterion/releases/download/$version/criterion-$version-linux-x86_64.tar.bz2
-  tar jxf criterion-$version-osx-x86_64.tar.bz2
-  tar jxf criterion-$version-linux-x86_64.tar.bz2
-  rm -rf criterion-$version-osx-x86_64.tar.bz2 criterion-$version-linux-x86_64.tar.bz2
+if [[ ! -d criterion-$version ]]
```
Comment (mvines) [inline programs/bpf/c/sdk/scripts/install.sh]: `if [[ ! -d criterion-$version ]]; then` please.
"then" gets super lonely when it's relegated to the next line :)

Comment (jackcmay) [reply programs/bpf/c/sdk/scripts/install.sh]: Such compassion!

**Comment by jackcmay** [comment]
@mvines, CI passed, any other feedback?  Anything I might have missed with the SDK, web3, etc...?

**Comment by mvines** [comment]
I think we want to include `scripts/install.sh` in the bpf-sdk tarball, it's currently removed at https://github.com/solana-labs/solana/blob/master/programs/bpf/c/sdk/scripts/package.sh#L17 and we don't want to run `install.sh` two lines above that.   

Instead we can make the downstream users run install.sh, so that'll be a change in solana-web3.js once this PR merges. 

There reason is LLVM is many MBs and I don't think we want those MBs packaged in the npmjs module

Beyond that the PR looks good to me

**Pull request merged**